recipes/magnum/all/conanfile.py

```diff
@@ -0,0 +1,710 @@
+        "fPIC": [True, False],
+        "shared_plugins": [True, False],
+
+        # Follow documented build-options in https://doc.magnum.graphics/magnum/building.html#building-features
+        
+        "target_gl": [True, False],
+        "target_gles": [True, False],
+        "target_gles2": [True, False],
+        "target_desktop_gles": [True, False],
+        "target_headless": [True, False],
+        "target_vk": [True, False],
+
+        "with_audio": [True, False],
+        "with_debugtools": [True, False],
+        "with_gl": [True, False],
+        "with_meshtools": [True, False],
+        "with_primitives": [True, False],
+        "with_scenegraph": [True, False],
+        "with_shaders": [True, False],
+        #"with_shaderstools": [True, False],  Option not available in sources!
+        "with_text": [True, False],
+        "with_texturetools": [True, False],
+        "with_trade": [True, False],
+        "with_vk": [True, False],
+        
+        "with_androidapplication": [True, False],
+        "with_emscriptenapplication": [True, False],
+        "with_glfwapplication": [True, False],
+        "with_glxapplication": [True, False],
+        "with_sdl2application": [True, False],
+        "with_xeglapplication": [True, False],
+        "with_windowlesscglapplication": [True, False],
+        "with_windowlesseglapplication": [True, False],
+        "with_windowlessglxapplication": [True, False],
+        "with_windowlessiosapplication": [True, False],
+        "with_windowlesswglapplication": [True, False],
+        "with_windowlesswindowseglapplication": [True, False],
+
+        "with_cglcontext": [True, False],
+        "with_eglcontext": [True, False],
+        "with_glxcontext": [True, False],
+        "with_wglcontext": [True, False],
+
+        "with_gl_info": [True, False],
+        # "with_vk_info": [True, False], Not in sources
+        "with_al_info": [True, False],
+        "with_distancefieldconverter": [True, False],
+        "with_fontconverter": [True, False],
+        "with_imageconverter": [True, False],
+        "with_sceneconverter": [True, False],
+        # "with_shaderconverter": [True, False],  Not in sources
+
+        # Options related to plugins
+        "with_anyaudioimporter": [True, False],
+        "with_anyimageconverter": [True, False],
+        "with_anyimageimporter": [True, False],
+        "with_anysceneconverter": [True, False],
+        "with_anysceneimporter": [True, False],
+        #"with_anyshaderconverter": [True, False],  Not in sources
+        "with_magnumfont": [True, False],
+        "with_magnumfontconverter": [True, False],
+        "with_objimporter": [True, False],
+        "with_tgaimporter": [True, False],
+        "with_tgaimageconverter": [True, False],
+        "with_wavaudioimporter": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "shared_plugins": True,
+
+        "target_gl": True,
```

Comment:
It looks like the names of the options are 1:1 copies of the cmake script.
I would rename these options such that they can be understood without reading the cmake script.

`target_gl` -> `enable_gl`
target_gles`, `target_gles` and `target_desktop_gles` can all be enabled (they are not not completely exclusive -> there exists a dependence). I would rename them to `enable_*`.

Reply:
About options, yes, they are 1-1 copies of CMake script. I really needed it while developing the recipe and figuring out all the logic. It was a bit hard sometimes :sweat:  I agree we can change them, letś try to choose the best names for them.

The project has options `with_gl` that enables GL component an requires one more dependency, but also `target_gl` that will add OpenGL support to other libraries (adds some extra sources). It is the same for options `with/target_vk`.

Reply:
About GL, I would say those options (gles, gles2 and desktop) are mutually exclusive and all require `with_gl`. This is from sources:

```cmake
if(WITH_GL)
    if(NOT TARGET_GLES OR TARGET_DESKTOP_GLES)
        # OpenGL library preference. Prefer to use GLVND, since that's the
        # better approach nowadays, but allow the users to override it from
        # outside in case it is broken for some reason (Nvidia drivers in
        # Debian's testing (Buster) -- reported on 2019-04-09).
        if(NOT CMAKE_VERSION VERSION_LESS 3.10 AND NOT OpenGL_GL_PREFERENCE)
            set(OpenGL_GL_PREFERENCE GLVND)
        endif()
        find_package(OpenGL REQUIRED)
    elseif(TARGET_GLES2)
        find_package(OpenGLES2 REQUIRED)
    else()
        find_package(OpenGLES3 REQUIRED)
    endif()
else()
    if(TARGET_GL)
        message(FATAL_ERROR "WITH_GL is OFF but TARGET_GL is enabled. Hmm.")
    endif()
    ...
```

Reply:
I think I will rename `with_xxxx` to `xxxx` (it says about the libraries that are built), but I feel like that the `target_xxx` should remain the same (they say about the _supported_ features in existing libraries)